Sources/Utilities/KMTextToSpeechHandler.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  KMTextToSpeechHandler.swift
+//  KommunicateChatUI-iOS-SDK
+//
+//  Created by sathyan elangovan on 29/07/22.
+//
+
+import Foundation
+import AVFoundation
+import KommunicateCore_iOS_SDK
+
+// To handle Text To Speech in the conversation
+class KMTextToSpeechHandler : NSObject, AVSpeechSynthesizerDelegate  {
+    let synthesizer : AVSpeechSynthesizer = AVSpeechSynthesizer()
+    var index = 0
+    var speechStarted = false
+   
+    public override init() {
+        super.init()
+        synthesizer.delegate = self
+    }
+    
+    public static let shared = KMTextToSpeechHandler()
+    var messageModels : [ALMessage] = []
+   
+    public func speechSynthesizer(_ synthesizer: AVSpeechSynthesizer, didFinish utterance: AVSpeechUtterance) {
+        index += 1
+        guard index < messageModels.count else {return speechStarted = false}
+        speakCurrentMessage()
+    }
+    
+    /**
+         This method is to trigger the specch from message
+        - Parameters:
+        - triggeredEvent : event type
+        - data : data of triggered event
+     */
+    func speakCurrentMessage(){
+         guard index < messageModels.count else {return speechStarted = false}
+         let utterance = AVSpeechUtterance(string: messageModels[index].message ?? "")
+         synthesizer.speak(utterance)
+    }
+    
+    /**
+    This method is to add message to existing list for the specch
+   - Parameters:
+   - list : [ALMessage]
+     */
+    func addMessagesToSpeech(_ list: [ALMessage]) {
+         for item in list {
+             if !messageModels.contains(item) {
```

Comment:
Will it compare message IDs? Otherwise we can remove it if there won't be any duplicate messages.

Reply:
removed the comparison.done